libkbfs/block_journal.go

```diff
@@ -72,18 +59,20 @@ type blockJournal struct {
 	log      logger.Logger
 	deferLog logger.Logger
 
-	j    diskJournal
-	refs map[BlockID]blockRefMap
+	// j is the main journal.
+	j diskJournal
 
+	// saveUntilMDFlush, when non-nil, prevents garbage collection
+	// of blocks. When removed, all the referenced blocks are
+	// garbage-collected.
+	//
+	// TODO: We only really need to save a list of IDs, and not a
+	// full journal.
 	saveUntilMDFlush *diskJournal
 
-	// Tracks the total size of on-disk blocks that will be flushed to
-	// the server (i.e., does not count reference adds).  It is only
-	// accurate for users of this journal that properly flush entries;
-	// in particular, calls to `removeReferences` with
-	// removeUnreferencedBlocks set to true can cause this count to
-	// deviate from the actual disk usage of the journal.
-	unflushedBytes int64
+	// s stores all the block data. s should always reflect the
+	// state you get by replaying all the entries in j.
+	s *blockDiskStore
```

Comment:
Why remove the `unflushedBytes` field entirely?  We should keep it around if only not to lose the comment, but I actually think it would be easy to restore (and not confuse people who are used to looking at the byte count in the status as a measure of progress).  Just use `filepath.Walk()` in the disk store and add up the bytes of the data files.  Should be way faster than the journal scan since it's only loading directory metadata, not reading any actual file data.